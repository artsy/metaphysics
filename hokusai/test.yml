version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
