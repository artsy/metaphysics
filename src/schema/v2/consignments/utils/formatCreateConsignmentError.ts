import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
