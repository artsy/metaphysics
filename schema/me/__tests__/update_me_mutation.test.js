import { runAuthenticatedQuery } from "test/utils"
import gql from "test/gql"

describe("UpdateMeMutation", () => {
  it("updates the user profile and returns its new data payload", () => {
    const mutation = gql`
      mutation {
        updateMyUserProfile(
          input: {
            collector_level: 1
            clientMutationId: "1232"
            phone: "1234890"
            location: { address: "123 my street" }
            price_range_min: -1
            price_range_max: 50000
          }
        ) {
          user {
            name
            phone
            location {
              city
              address
            }
            price_range
          }
        }
      }
    `

    const rootValue = {
      updateMeLoader: () =>
        Promise.resolve({
          id: "106",
          name: "andy-warhol",
          phone: "1234890",
          location: {
            address: "123 my street",
          },
          price_range: "-1:50000",
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
