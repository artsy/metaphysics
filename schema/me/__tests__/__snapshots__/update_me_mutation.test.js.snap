// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMeMutation updates the user profile and returns its new data payload 1`] = `
Object {
  "updateMyUserProfile": Object {
    "user": Object {
      "location": Object {
        "address": "123 my street",
        "city": null,
      },
      "name": "andy-warhol",
      "phone": "1234890",
    },
  },
}
`;
